buildIncremental.fsx

```diff
@@ -42,18 +44,37 @@ module IncrementalTests =
         | EndsWith "cmd" -> true
         | EndsWith "sh" -> true
         | _ -> false
+    
+    let getHeadHashFor repositoryDir branch =
+        let _, msg, error = runGitCommand repositoryDir (sprintf "log --oneline -1 %s" branch)
+        if error <> "" then failwithf "git log --oneline failed: %s" error
+        let logMsg = msg |> Seq.head
+        let tmp =
+            logMsg.Split(' ')
+            |> Seq.head
+            |> fun s -> s.Split('m')
+        if tmp |> Array.length > 2 then tmp.[1].Substring(0,6) else tmp.[0].Substring(0,6)
```

Comment:
What exactly is this doing? I know the function is trying to get the hash head.... Is this just trying to get the first six characters of the hash?